foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiClient.java

```diff
@@ -58,7 +58,11 @@ public final class OpenAiClient {
    */
   @Nonnull
   public static OpenAiClient forModel(@Nonnull final OpenAiModel foundationModel) {
-    final var destination = Core.getDestinationForModel(foundationModel.model(), "default");
+    final var destination =
+        new AiCoreService()
+            .forDeploymentByModel(foundationModel.model())
+            .withResourceGroup("default")
+            .destination();
```

Comment:
It might be useful to store the AI Core service itself, not the destination. That way we get the destination on demand, which might be useful if we ever need the destination to change.

Reply:
Agree, but I would rather change this (semi-related) improvement in a dedicated PR.